api/apps/api/src/modules/planning-units/planning-units.service.ts

```diff
@@ -1,79 +1,32 @@
-import { FactoryProvider, Inject, Injectable } from '@nestjs/common';
+import { Inject, Injectable } from '@nestjs/common';
 
-import { Job, Queue, QueueEvents } from 'bullmq';
-import { ApiEventsService } from '@marxan-api/modules/api-events';
+import { Queue } from 'bullmq';
 import { API_EVENT_KINDS } from '@marxan/api-events';
-import {
-  PlanningUnitsJob,
-  createQueueName,
-} from '@marxan-jobs/planning-unit-geometry';
-import { QueueBuilder, QueueEventsBuilder } from '@marxan-api/modules/queue';
-
-export const queueToken = Symbol(`planning unit queue token`);
-export const queueProvider: FactoryProvider<Queue<PlanningUnitsJob, void>> = {
-  provide: queueToken,
-  useFactory: (builder: QueueBuilder) => builder.buildQueue(createQueueName),
-  inject: [QueueBuilder],
-};
-export const queueEventsToken = Symbol(`planning units queue events token`);
-export const queueEventsProvider: FactoryProvider<QueueEvents> = {
-  provide: queueEventsToken,
-  useFactory: (builder: QueueEventsBuilder) =>
-    builder.buildQueueEvents(createQueueName),
-  inject: [QueueEventsBuilder],
-};
+import { ApiEventsService } from '@marxan-api/modules/api-events';
+import { PlanningUnitsJob } from '@marxan-jobs/planning-unit-geometry';
 
-type CompletedEvent = { jobId: string; returnvalue: string };
-type FailedEvent = { jobId: string; failedReason: string };
+import { queueToken } from './planning-units-queue.provider';
 
 @Injectable()
 export class PlanningUnitsService {
   constructor(
     @Inject(queueToken)
     private readonly queue: Queue<PlanningUnitsJob, void>,
-    @Inject(queueEventsToken)
-    private readonly queueEvents: QueueEvents,
     private readonly events: ApiEventsService,
-  ) {
-    this.queueEvents.on(`completed`, (data, eventId) =>
-      this.onCompleted(data, eventId),
-    );
-    this.queueEvents.on(`failed`, (data, eventId) =>
-      this.onFailed(data, eventId),
-    );
-  }
+  ) {}
 
-  public async create(jobDefinition: PlanningUnitsJob): Promise<void> {
-    await this.queue.add('create-regular-pu', jobDefinition);
-    await this.events.create({
-      kind: API_EVENT_KINDS.project__planningUnits__submitted__v1__alpha,
-      topic: jobDefinition.projectId,
-    });
-  }
+  async create(jobDefinition: PlanningUnitsJob) {
+    const kind = API_EVENT_KINDS.project__planningUnits__submitted__v1__alpha;
+    const job = await this.queue.add(`create-regular-pu`, jobDefinition);
 
-  private async onCompleted(data: CompletedEvent, eventId: string) {
-    const job: Job<PlanningUnitsJob> | undefined = await this.queue.getJob(
-      data.jobId,
-    );
-    if (!job) return;
-    const { projectId } = job.data;
-    await this.events.createIfNotExists({
-      kind: API_EVENT_KINDS.project__planningUnits__finished__v1__alpha,
-      topic: projectId,
-      externalId: eventId,
-    });
-  }
+    // bad typing - may happen that job wasn't added
+    if (!job) {
+      return;
```

Comment:
I'd suggest adding some logging here - if adding jobs starts to fail for some reason, we'd want to easily be able to track where that happens

Reply:
@ikas Added in https://github.com/Vizzuality/marxan-cloud/pull/643/commits/bedc454297d448da6ba67299a25837e7b9cd1058

Reply:
nice :ok_hand: